templates/widgets/view.html.twig

```diff
@@ -0,0 +1,41 @@
+<h4>{{ item.name }}</h4>
+<p>
+    Created by: {{ item.user }}
+</p>
+<div id="widget-data-container"></div>
+
+<!-- Modal -->
+{# <div class="modal fade" id="widget_view" aria-hidden="true">
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h5 class="modal-title" id="modalWidgetLabel">Modal title</h5>
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+            </div>
+            <div class="modal-body">
+
+            </div>
+            <div class="modal-footer">
+                <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+            </div>
+        </div>
+    </div>
+</div> #}
+
+<script>
+    document.addEventListener('DOMContentLoaded', function() {
+       // alert("Ready!");
+        $(function(){
+        $.ajax('/widget/generate_diagram',{
+            method: 'post',
+            data: JSON.parse('{{ item.widget|raw }}'),
```

Comment:
The best will be if we define a JavaScript variable called widgetJson = item.widget|raw and we pass it to AJAX data. By doing this we don't need to have JSON.parse()

This is an example where it fails for me:

```
{
                title: 'Device Distribution',
                global: {
                    presentation: {
                        'render_type': 'distribution'
                    },
                    data_requirements: {
                        period: {
                            type: 'moving',
                            start: '30',
                            end: 'now',
                        }
                    }
                },
                segments: {
                    1: {
                        presentation: {
                            name: 'Desktop',
                            color: '#1F77B4'
                        },
                        data_requirements: {
                            filters: {
                                device_type: {
                                    search_value: 2,
                                    condition: 'is'
                                }
                            },
                            business_metrics: {
                                page_views_count: {
                                    data_flavor: {
                                        count: true
                                    }
                                }
                            }
                        }
                    },
                    2: {
                        presentation: {
                            name: 'Tablet',
                            color: '#ff6023'
                        },
                        data_requirements: {
                            filters: {
                                device_type: {
                                    search_value: 3,
                                    condition: 'is'
                                }
                            },
                            business_metrics: {
                                page_views_count: {
                                    data_flavor: {
                                        count: true
                                    }
                                }
                            }
                        }
                    },
                    3: {
                        presentation: {
                            name: 'Mobile',
                            color: '#2CA02C',
                        },
                        data_requirements: {
                            filters: {
                                device_type: {
                                    search_value: 1,
                                    condition: 'is'
                                }
                            },
                            business_metrics: {
                                page_views_count: {
                                    data_flavor: {
                                        count: true
                                    }
                                }
                            }
                        }
                    },
                    4: {
                        presentation: {
                            name: 'Bot',
                            color: '#000000'
                        },
                        data_requirements: {
                            filters: {
                                device_type: {
                                    search_value: 4,
                                    condition: 'is'
                                }
                            },
                            business_metrics: {
                                page_views_count: {
                                    data_flavor: {
                                        count: true
                                    }
                                }
                            }
                        }
                    }
                }
            }
```